platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/logging/HtmlTagFactory.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.cli.logging;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class for constructing HTML tags.
+ */
+public class HtmlTagFactory {
+    /**
+     * Map from attribute name to list of attribute values.
+     */
+    private final Map<String, List<String>> attributeMap = new HashMap<>();
+
+    /**
+     * The type of HTML tag
+     */
+    private final String tagName;
+
+    /**
+     * The content of the HTML tag. May be empty
+     */
+    private final String content;
+
+    /**
+     * If true then the tag is a void tag
+     */
+    private final boolean isVoidTag;
+
+    /**
+     * Construct a new HtmlTagFactory.
+     *
+     * @param tagName The type of HTML tag
+     * @param content The content of the HTML tag. Must be null if voidTag is true
+     * @param voidTag If the tag is a void tag
+     */
+    public HtmlTagFactory(@NonNull final String tagName, @Nullable final String content, final boolean voidTag) {
+        this.tagName = Objects.requireNonNull(tagName);
+
+        if (voidTag && content != null) {
+            throw new IllegalArgumentException("content must be null if voidTag is true");
+        }
+
+        this.content = content;
+        this.isVoidTag = voidTag;
+    }
```

Comment:
Perhaps instead of the `voidTag` argument, create a second constructor with just the tag name.